src/com/twu/biblioteca/view/PromptView.java

```diff
@@ -0,0 +1,45 @@
+package com.twu.biblioteca.view;
+
+import java.util.Scanner;
+import java.util.function.Function;
+
+public abstract class PromptView implements View {
+    private Scanner reader;
+
+    public PromptView(Scanner reader) {
+        this.reader = reader;
+    }
+
+    public String ask(String question) {
+        show(question);
+        return reader.nextLine();
+    }
+
+    public <T> T askAndParse(String question, Function<String, T> parser) {
+        show(question);
+        return parser.apply(reader.nextLine());
+    }
+
+    public void show(String message) {
+        System.out.println(message);
+    }
+
+    public <T extends View> void goTo(Class<T> view) {
+        try {
+            View newView = view.getConstructor(Scanner.class).newInstance(reader);
+            newView.initialize();
```

Comment:
Muito interessante essa tua ideia para gerenciar a navegação entre telas. Bom uso das ferramentas de reflexão da linguagem :)
Eu não analisei a fundo, mas é possível que a tua implementação tenha um pequeno problema de gerência de memória e pilha. Corrija-me se eu estiver enganado, mas basicamente tu vai empilhando chamadas de goTo que nunca resolvem (cada chamada de goTo é finalzada por uma nova chamada de goTo). Se for esse o caso, tu nunca nunca remove as chamadas da pilha e os objetos alocados nunca são coletados pelo garbage collector. Navegar vezes o suficiente pode estourar a pilha ou o heap da JVM.

Reply:
Se quiser brincar com isso, qualquer hora podemos fazer profiling da tua aplicação usando a interface JMX pra monitorar o heap e pilha, tirando essa dúvida ae 😃

Reply:
Eita que massa! Não fazia ideia disso, daora bora fazer sim!